impl/src/main/java/org/jboss/weld/util/Proxies.java

```diff
@@ -81,7 +81,7 @@ private TypeInfo(Set<? extends Type> types) {
         private TypeInfo add(Type type, List<Class<?>> foundInterfaces, List<Class<?>> foundClasses, Map<String, String> classToPackageMap) {
             if (type instanceof Class<?>) {
                 Class<?> clazz = (Class<?>) type;
-                classToPackageMap.put(clazz.getName(), clazz.getPackage().getName());
+                classToPackageMap.put(clazz.getName(), clazz.getPackage()==null?null:clazz.getPackage().getName());
```

Comment:
While this should be technically safe (since the package should then be replaced), I'd rather we store empty String in the map instead of just `null`.